src/app/modules/form/admin-form/__tests__/admin-form.controller.spec.ts

```diff
@@ -6564,4 +6566,221 @@ describe('admin-form.controller', () => {
+    const mockFormLogic = {
+      form_logics: [
+        {
+          _id: logicId,
+          id: logicId,
+        } as ILogicSchema,
+      ],
+    }
+
+    const MOCK_FORM = {
+      admin: MOCK_USER,
+      _id: MOCK_FORM_ID,
+      title: 'mock title',
+      ...mockFormLogic,
+    } as IPopulatedForm
+
+    const mockReq = expressHandler.mockRequest({
+      params: {
+        formId: MOCK_FORM_ID,
+        logicId: logicId,
+      },
+      session: {
+        user: {
+          _id: MOCK_USER_ID,
+        },
+      },
+    })
+    const mockRes = expressHandler.mockResponse()
+
+    beforeEach(() => {
+      MockUserService.getPopulatedUserById.mockReturnValue(okAsync(MOCK_USER))
+      MockAuthService.getFormAfterPermissionChecks.mockReturnValue(
+        okAsync(MOCK_FORM),
+      )
+      MockAdminFormService.deleteFormLogic.mockReturnValue(okAsync(true))
+    })
+
+    it('should call all services correctly when request is valid', async () => {
+      await AdminFormController.handleDeleteLogic(mockReq, mockRes, jest.fn())
+
+      expect(MockUserService.getPopulatedUserById).toHaveBeenCalledWith(
+        MOCK_USER_ID,
+      )
+      expect(MockAuthService.getFormAfterPermissionChecks).toHaveBeenCalledWith(
+        {
+          user: MOCK_USER,
+          formId: MOCK_FORM_ID,
+          level: PermissionLevel.Write,
+        },
+      )
+      expect(MockAdminFormService.deleteFormLogic).toHaveBeenCalledWith(
+        MOCK_FORM,
+        logicId,
+      )
+
+      expect(mockRes.json).toHaveBeenCalledWith({
+        message: 'Logic deleted successfully',
+      })
+    })
+
+    it('should return 403 when user does not have permissions to delete logic', async () => {
+      MockAuthService.getFormAfterPermissionChecks.mockReturnValue(
+        errAsync(
+          new ForbiddenFormError('not authorized to perform write operation'),
+        ),
+      )
+
+      await AdminFormController.handleDeleteLogic(mockReq, mockRes, jest.fn())
+
+      expect(MockUserService.getPopulatedUserById).toHaveBeenCalledWith(
+        MOCK_USER_ID,
+      )
+      expect(MockAuthService.getFormAfterPermissionChecks).toHaveBeenCalledWith(
+        {
+          user: MOCK_USER,
+          formId: MOCK_FORM_ID,
+          level: PermissionLevel.Write,
+        },
+      )
+      expect(MockAdminFormService.deleteFormLogic).not.toHaveBeenCalled()
+
+      expect(mockRes.status).toHaveBeenCalledWith(403)
+
+      expect(mockRes.json).toHaveBeenCalledWith({
+        message: 'not authorized to perform write operation',
+      })
+    })
+
+    it('should return 404 when logicId cannot be found', async () => {
+      const wrongLogicId = new ObjectId().toHexString()
+      const mockReqWrongLogic = expressHandler.mockRequest({
+        params: {
+          formId: MOCK_FORM_ID,
+          logicId: wrongLogicId,
+        },
+        session: {
+          user: {
+            _id: MOCK_USER_ID,
+          },
+        },
+      })
```

Comment:
this isn't necessary right, since you've mocked `deleteFormLogic`?

Reply:
yes, it isn't necessary in order to test `deleteFormLogic`, but since we're testing `handleDeleteLogic`, this allows us to test that `handleDeleteLogic` passes the correct param value for `logicId` into `deleteFormLogic`

Reply:
but you've already defined `mockReq` right? so you just need to test that `deleteFormLogic` is called with `logicId` instead of `wrongLogicId`. just trying to streamline the testing code here.

Reply:
ok, no strong views here. edited.